cmds/cpud/serve.go

```diff
@@ -115,6 +125,31 @@ func serve() error {
 	if err != nil {
 		return err
 	}
+	// TODO: This is overly restrictive
```

Comment:
This should be done in the server package.

Reply:
Could we indicate mdns in the network address somehow and let client code figure it out? 
What if as part of hostname parsing we could just magically now on the client side what to do? What if on server side we could, similarly, know that cpud needs to be an mdns server? There seem to be some kind of standards out there for it. I really want to get this in, it would be super handy, just trying to work out how. 

Can we just mount an mdns server on /net/cs -- oh wait ...

Reply:
Yeah, I did think that we could do magic hostname specification -- of course that always comes with its own caveats.  Server side autodetecting if it has to act as server seems a bit more dodgey - in general things seem to work on metal, but docker won't let multiple people control the port.  There seem to be multiple things out there to make docker play nice with mdns, but haven't had time to really try any of them in anger.  In any case, probably will need to make that different variations of options (host your own server, play nice with dbus, use our own mdns port, do something else)

Sorry - was a busy week this last week, so I didn't get around to doing much here - was planning on spending some time this week implementing the code reorg into the modules like you suggested and fleshing out the meta-data based masking and prioritization.

Reply:
very cool, I was surprised to see this closed but if you're going to keep at it, we REALLY need it. It's going to make life a lot better for our use cases.

Reply:
server portion migrated to server module, but it wasn't as clean as I would like.  Suggestions welcome.